source/common/upstream/health_discovery_service.cc

```diff
@@ -102,27 +102,51 @@ void HdsDelegate::handleFailure() {
 // TODO(lilika): Add support for the same endpoint in different clusters/ports
 envoy::service::health::v3::HealthCheckRequestOrEndpointHealthResponse HdsDelegate::sendResponse() {
   envoy::service::health::v3::HealthCheckRequestOrEndpointHealthResponse response;
+
   for (const auto& cluster : hds_clusters_) {
+    // Add cluster health response and set name.
+    auto* cluster_health =
+        response.mutable_endpoint_health_response()->add_cluster_endpoints_health();
+    cluster_health->set_cluster_name(cluster->info()->name());
+
+    // Iterate through all hosts in our priority set.
     for (const auto& hosts : cluster->prioritySet().hostSetsPerPriority()) {
-      for (const auto& host : hosts->hosts()) {
-        auto* endpoint = response.mutable_endpoint_health_response()->add_endpoints_health();
-        Network::Utility::addressToProtobufAddress(
-            *host->address(), *endpoint->mutable_endpoint()->mutable_address());
-        // TODO(lilika): Add support for more granular options of
-        // envoy::config::core::v3::HealthStatus
-        if (host->health() == Host::Health::Healthy) {
-          endpoint->set_health_status(envoy::config::core::v3::HEALTHY);
-        } else {
-          if (host->getActiveHealthFailureType() == Host::ActiveHealthFailureType::TIMEOUT) {
-            endpoint->set_health_status(envoy::config::core::v3::TIMEOUT);
-          } else if (host->getActiveHealthFailureType() ==
-                     Host::ActiveHealthFailureType::UNHEALTHY) {
-            endpoint->set_health_status(envoy::config::core::v3::UNHEALTHY);
-          } else if (host->getActiveHealthFailureType() == Host::ActiveHealthFailureType::UNKNOWN) {
-            endpoint->set_health_status(envoy::config::core::v3::UNHEALTHY);
+      // Get a grouping of hosts by locality.
+      for (const auto& locality_hosts : hosts->hostsPerLocality().get()) {
+
```

Comment:
Nit: avoid superfluous spaces that don't provide structure hinting, Envoy prefers compact code.

Reply:
I see, thanks! These white space issues are now fixed.